util/util.go

```diff
@@ -37,28 +35,11 @@ var HostportPattern = regexp.MustCompile(`^(\d+.\d+.\d+.\d+):\d+$`)
 
 // CaptureHost takes a host:port and returns the host.
 func CaptureHost(hostport string) string {
-	if HostportPattern.Match([]byte(hostport)) {
-		parts := strings.Split(hostport, ":")
-		return parts[0]
-	}
-	return ""
-}
-
-// ReadHostsFile reads a file containing a JSON array of hosts
-func ReadHostsFile(file string) ([]string, error) {
-	var hosts []string
-
-	data, err := ioutil.ReadFile(file)
-	if err != nil {
-		return nil, err
-	}
-
-	err = json.Unmarshal(data, &hosts)
+	host, _, err := net.SplitHostPort(hostport)
 	if err != nil {
-		return nil, err
+		return ""
```

Comment:
Actually, I rescind this comment. I started pulling the thread here but `CaptureHost` is used in some pretty important methods and it opens up questions about error handling which then affects exported methods.

I think we should leave this one as-is for now.